M3 6045: Add modal content, tabs and validation check

cpathipa opened the pull request:
## Description 📝
Show the API awareness modal after validation check passes.

**What does this PR do?**

- Adds content and respective links to the API awareness modal.
- Adds Tabs component.
- Add installation guides.
- Wired GA events
-  Validation check to show / hide modal

## Preview 📷
![image](https://user-images.githubusercontent.com/119517080/212408858-adb3aeac-6951-457e-89a9-1c038594e281.png)


## How to test 🧪

- Navigate to Create Linode page
- Click on "Create using command line" button - Try with out filling required data.
- Should show the inline error messages.
- Enter data for all required fields.
- Click on "Create using command line"
- Show show the API awareness modal.
- Validate all the links. 

**How do I run relevant unit or e2e tests?**
yarn test

cpathipa commented on packages/manager/src/components/Notice/Notice.tsx:
```diff
@@ -132,6 +133,7 @@ interface Props extends GridProps {
   errorGroup?: string;
   important?: boolean;
   warning?: boolean;
+  marketing?: boolean;
```
Named it as `marketing`  after referring [design.linode.com](https://design.linode.com/?path=/docs/components-notifications-notices--success)

hana-linode replied:
We're actually not using green for marketing anymore but blue instead. So you can use the success variant instead of defining a new variant

I updated the Notice story in my [banner and notices PR](https://github.com/linode/manager/pull/8714) to reflect the change from green to blue

cpathipa replied:
Thank you @hana-linode!

cpathipa commented on packages/manager/src/themeFactory.ts:
```diff
@@ -392,6 +393,7 @@ const themeDefaults: ThemeDefaults = () => {
         warningDark: '#ffd002',
         error: '#f8dedf',
         errorDark: '#cd2227',
+        marketingDark: '#00b159',
```
Picked closest color from [design.linode,com](https://design.linode.com/?path=/docs/core-styles-color-palette--color-palette)

hana-linode commented on packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx:
```diff
@@ -4,19 +4,65 @@ import ActionsPanel from 'src/components/ActionsPanel';
+import TabLinkList from 'src/components/TabLinkList';
 import Typography from 'src/components/core/Typography';
+import Notice from 'src/components/Notice';
+
```
Can we remove the empty lines between imports?

hana-linode reviewed (commented):
Can we make the dialog height smaller/scrollable so it doesn't take up the whole screen?

![image](https://user-images.githubusercontent.com/115299789/212176084-b4c6a42e-a3a0-4c78-84aa-429730b36db6.png)

dwiley-akamai reviewed (commented):
The functionality is mostly in place and the UI mostly matches the mockups but I noticed a few things while testing:

- The `Check out our collection of integrations ...` notice appears to be bold in the mockups
- Can we some top padding to create more space between the "cURL" / "Linode CLI" tabbed section and the modal instruction text above it? (per the mockups)
- Right now it seems that we're unable to use keyboard navigation to switch between the "cURL" and "Linode CLI" tabs
- If you fill in all the required fields and open the modal, but then close the modal and clear out one or more of the required fields, you're able to still click "Create Using Command Line" and successfully open the modal.

dwiley-akamai commented:
> Can we make the dialog height smaller/scrollable so it doesn't take up the whole screen?

I believe this is our standard practice for dialogs (the Resize, Rescue, Migrate, etc. dialogs are all full size)

cpathipa commented:
> The functionality is mostly in place and the UI mostly matches the mockups but I noticed a few things while testing:
> 
> * The `Check out our collection of integrations ...` notice appears to be bold in the mockups
> * Can we some top padding to create more space between the "cURL" / "Linode CLI" tabbed section and the modal instruction text above it? (per the mockups)
> * Right now it seems that we're unable to use keyboard navigation to switch between the "cURL" and "Linode CLI" tabs
> * If you fill in all the required fields and open the modal, but then close the modal and clear out one or more of the required fields, you're able to still click "Create Using Command Line" and successfully open the modal.

Great feedback @dwiley-akamai @hana-linode Addressed above feedback and PR is ready for review.
- As we got some design changes and specs, bottom `notice` content will be part of it. We have story(6606) to cover that along with other language changes.
- Added styes ( for spacing) as suggested by Andrew.
-  As I was able to navigate thru tabs. I think we are using common component for Tabs. If you still have issues, lmk, we can pair on that. (Attached screen recording for reference)
- As we discussed, Image field is not required to create Linode. So we are covered for the fourth point. 

- Reduced model height

https://user-images.githubusercontent.com/119517080/212423555-fd14a7c5-5e72-44a1-8cda-c269b576fb8c.mov

dwiley-akamai reviewed (commented):
Just confirmed that the keyboard navigation with the tabs is working properly, not sure what the issue was yesterday. I'll complete a full review of this on Tuesday 👍

mjac0bs reviewed (commented):
Leaving some comments on copy, spacing, and linkage, then I realized that you may already plan on addressing these in a follow up ticket (M3-6067), but I wasn't certain.

I also noticed the Linode plan selection table is defaulting the Shared CPU rather than Default CPU for the plan type and we'll want to make sure this is rebased on `develop` given the bugfix last week.

bnussman-akamai commented on packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx:
```diff
@@ -631,6 +642,23 @@ class LinodeCreateContainer extends React.PureComponent<CombinedProps, State> {
+    try {
+      CreateLinodeSchema.validateSync(payload, { abortEarly: false });
+      //reset errors to default state
+      this.setState({ errors: undefined, showApiAwarenessModal: true });
```
Do we need `showApiAwarenessModal: true` here?

```suggestion
      this.setState({ errors: undefined });
```

cpathipa replied:
@bnussman-akamai Yes, I think we need it to show the Modal. Are you able to render the modal without `showApiAwarenessModal`  ?

cpathipa replied:
Since we need `showApiAwarenessModal` to show the modal. Hence marking it as resolved.

bnussman-akamai commented on packages/manager/src/features/linodes/LinodesCreate/LinodeCreateContainer.tsx:
```diff
@@ -404,6 +410,12 @@ class LinodeCreateContainer extends React.PureComponent<CombinedProps, State> {
     }));
   };
 
+  handleShowApiAwarenessModal = () => {
```
Consider updating the function name to better match the behavior

```suggestion
  handleToggleApiAwarenessModal = () => {
```

dwiley-akamai commented on packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx:
```diff
@@ -1,51 +1,148 @@
+      </Notice>
 
-      <ActionsPanel>
+      <ActionsPanel style={{ marginTop: '0px', paddingTop: '4px' }}>
```
Can we place this styling in a `useStyles` class?

dwiley-akamai commented on packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx:
```diff
@@ -1,51 +1,148 @@
+        style={{ marginTop: '14px' }}
+      >
+        <TabLinkList tabs={tabs} />
+        <TabPanels style={{ paddingBottom: '24px', paddingTop: '16px' }}>
```
Can we place this styling in a `useStyles` class?

cpathipa commented on packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx:
```diff
@@ -1,51 +1,148 @@
+      <ActionsPanel style={{ marginTop: '0px', paddingTop: '4px' }}>
         <Button
           buttonType="secondary"
           onClick={onClose}
```
Good catch! @mjac0bs I totally missed it, Adjusted the spacing for `Close` button to align right. 
![image](https://user-images.githubusercontent.com/119517080/213174515-03a42dd5-539d-4af8-b15c-7559178b4fb7.png)

cpathipa commented on packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx:
```diff
@@ -1,51 +1,148 @@
         />{' '}
         then save your existing token to an environment variable or substitue it
         into the command. Read our guides to learn about creating{' '}
         <ExternalLink
```
@mjac0bs UX team confirmed that links should not have icons. Sorry for the confusion I should have updated in the PR description.

cpathipa commented on packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx:
```diff
@@ -1,51 +1,148 @@
-      <Typography variant="body1">
+      <Typography variant="body1" style={{ paddingTop: '16px' }}>
         You&#39;ll first need to{' '}
         <ExternalLink
```
@mjac0bs Technically we should use `Link` since we are using relative path. But, it causing user to lose the context. So, to overcome that we are using `ExternalLink`. ([context](https://github.com/linode/manager/pull/8689#discussion_r1063783542) )

cpathipa commented on packages/manager/src/features/linodes/LinodesCreate/ApiAwarenessModal/index.tsx:
```diff
@@ -1,51 +1,148 @@
+
 const ApiAwarenessModal = (props: Props) => {
-  const { isOpen, onClose } = props;
+  const { isOpen, label, onClose, route } = props;
```
@carrillo-erik I think we have similar naming conventions followed across the repo. I would say this could be good topic in frontend cafe.

cpathipa commented:
> Leaving some comments on copy, spacing, and linkage, then I realized that you may already plan on addressing these in a follow up ticket (M3-6067), but I wasn't certain.
> 
> I also noticed the Linode plan selection table is defaulting the Shared CPU rather than Default CPU for the plan type and we'll want to make sure this is rebased on `develop` given the bugfix last week.

@mjac0bs Once it is merged to feature branch api-cli, we can rebase the feature branch.

mjac0bs reviewed (approved):
LGTM now. Thanks for following up with those responses about the `Link` vs `ExternalLink` and aligning the close button.

cpathipa commented:
Great feedback @bnussman-akamai @carrillo-erik @dwiley-akamai @mjac0bs.
